test/spawn.jl

```diff
@@ -173,14 +173,17 @@ exename = joinpath(JULIA_HOME, Base.julia_exename())
 if valgrind_off
     # If --trace-children=yes is passed to valgrind, we will get a
     # valgrind banner here, not "Hello World\n".
-    @test readall(pipe(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    @test readall(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    p = Pipe()
+    run(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr = p))
```

Comment:
here's an even better example of why we should never export `PipeEnd` to the user:
```
p = PipeEnd()
spawn(pipeline(`foo` & `bar`, stdout=p)) # this expression has undefined behavior, since bar may get the wrong end of the pipe
```

(i'm most of the way done with a cleanup effort to add a `Pipe` object which tracks a pair of pipe ends)